src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = newNode;
+            last = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
```

Comment:
if clause may be replaced with `checkIndex` method you already have.

Reply:
> if clause may be replaced with `checkIndex` method you already have.

I can't replace it there different checks but I created a new method